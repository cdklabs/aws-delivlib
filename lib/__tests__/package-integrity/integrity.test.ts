import { SynthUtils } from '@monocdk-experiment/assert';
import '@monocdk-experiment/assert/jest';
import {
  App, Stack,
  aws_codebuild as codebuild,
  aws_secretsmanager as sm,
} from 'monocdk';
import { PackageIntegrityValidation } from '../..';

test('minimal snapshot', () => {

  const stack = new Stack(new App(), 'TestStack');

  const token = sm.Secret.fromSecretCompleteArn(stack, 'GitHubSecret', 'arn:aws:secretsmanager:us-east-1:123456789123:secret:github-token-000000');

  new PackageIntegrityValidation(stack, 'Integrity', {
    buildImage: codebuild.LinuxBuildImage.fromDockerRegistry('jsii/superchain:1-buster-slim-node12'),
    githubTokenSecret: token,
    repository: 'cdklabs/some-repo',
  });

  expect(SynthUtils.synthesize(stack).template).toMatchSnapshot();
});
