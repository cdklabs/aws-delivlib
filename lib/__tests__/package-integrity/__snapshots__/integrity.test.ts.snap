// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minimal snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806ArtifactHash31DC7E03": Object {
      "Description": "Artifact hash for asset \\"f60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806\\"",
      "Type": "String",
    },
    "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3BucketF65C1459": Object {
      "Description": "S3 bucket for asset \\"f60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806\\"",
      "Type": "String",
    },
    "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3VersionKey55608E43": Object {
      "Description": "S3 key for asset version \\"f60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "IntegrityScheduledTrigger3EAD0EE5": Object {
      "Properties": Object {
        "ScheduleExpression": "rate(1 day)",
        "State": "ENABLED",
        "Targets": Array [
          Object {
            "Arn": Object {
              "Fn::GetAtt": Array [
                "IntegrityValidate7B3C708F",
                "Arn",
              ],
            },
            "Id": "Target0",
            "RoleArn": Object {
              "Fn::GetAtt": Array [
                "IntegrityValidateEventsRole0B01F9FA",
                "Arn",
              ],
            },
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "IntegrityValidate7B3C708F": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "NO_ARTIFACTS",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_MEDIUM",
          "EnvironmentVariables": Array [
            Object {
              "Name": "SCRIPT_S3_BUCKET",
              "Type": "PLAINTEXT",
              "Value": Object {
                "Ref": "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3BucketF65C1459",
              },
            },
            Object {
              "Name": "SCRIPT_S3_KEY",
              "Type": "PLAINTEXT",
              "Value": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::Select": Array [
                        0,
                        Object {
                          "Fn::Split": Array [
                            "||",
                            Object {
                              "Ref": "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3VersionKey55608E43",
                            },
                          ],
                        },
                      ],
                    },
                    Object {
                      "Fn::Select": Array [
                        1,
                        Object {
                          "Fn::Split": Array [
                            "||",
                            Object {
                              "Ref": "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3VersionKey55608E43",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                ],
              },
            },
            Object {
              "Name": "GITHUB_REPOSITORY",
              "Type": "PLAINTEXT",
              "Value": "cdklabs/some-repo",
            },
            Object {
              "Name": "TAG_PREFIX",
              "Type": "PLAINTEXT",
              "Value": "",
            },
            Object {
              "Name": "GITHUB_TOKEN",
              "Type": "SECRETS_MANAGER",
              "Value": "github-token",
            },
          ],
          "Image": "jsii/superchain:1-buster-slim-node12",
          "ImagePullCredentialsType": "SERVICE_ROLE",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "IntegrityValidateRole785947FC",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"pre_build\\": {
      \\"commands\\": [
        \\"echo \\\\\\"Downloading scripts from s3://\${SCRIPT_S3_BUCKET}/\${SCRIPT_S3_KEY}\\\\\\"\\",
        \\"aws s3 cp s3://\${SCRIPT_S3_BUCKET}/\${SCRIPT_S3_KEY} /tmp\\",
        \\"mkdir -p /tmp/scriptdir\\",
        \\"unzip /tmp/$(basename $SCRIPT_S3_KEY) -d /tmp/scriptdir\\"
      ]
    },
    \\"build\\": {
      \\"commands\\": [
        \\"export SCRIPT_DIR=/tmp/scriptdir\\",
        \\"echo \\\\\\"Running validate.sh\\\\\\"\\",
        \\"/bin/bash /tmp/scriptdir/validate.sh\\"
      ]
    }
  }
}",
          "Type": "NO_SOURCE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "IntegrityValidateAlarmF5ED9E32": Object {
      "Properties": Object {
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "Dimensions": Array [
          Object {
            "Name": "ProjectName",
            "Value": Object {
              "Ref": "IntegrityValidate7B3C708F",
            },
          },
        ],
        "EvaluationPeriods": 1,
        "MetricName": "FailedBuilds",
        "Namespace": "AWS/CodeBuild",
        "Period": 300,
        "Statistic": "Sum",
        "Threshold": 1,
        "TreatMissingData": "ignore",
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "IntegrityValidateEventsRole0B01F9FA": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "events.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "IntegrityValidateEventsRoleDefaultPolicy369B5593": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "codebuild:StartBuild",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "IntegrityValidate7B3C708F",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "IntegrityValidateEventsRoleDefaultPolicy369B5593",
        "Roles": Array [
          Object {
            "Ref": "IntegrityValidateEventsRole0B01F9FA",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "IntegrityValidateRole785947FC": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "IntegrityValidateRoleDefaultPolicy9F4C8581": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":secretsmanager:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":secret:github-token-??????",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "IntegrityValidate7B3C708F",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "IntegrityValidate7B3C708F",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "IntegrityValidate7B3C708F",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3BucketF65C1459",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      Object {
                        "Ref": "AssetParametersf60150cbac8b2492682640e5871cb7353dbe98240f425a7efaa9f3cfe58bf806S3BucketF65C1459",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": "arn:aws:secretsmanager:us-east-1:123456789123:secret:github-token-000000",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "IntegrityValidateRoleDefaultPolicy9F4C8581",
        "Roles": Array [
          Object {
            "Ref": "IntegrityValidateRole785947FC",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
